Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp

```diff
@@ -328,6 +330,22 @@ void MediaStreamTrack::getPhotoCapabilities(DOMPromiseDeferred<IDLDictionary<Pho
     });
 }
 
+void MediaStreamTrack::getPhotoSettings(DOMPromiseDeferred<IDLDictionary<PhotoSettings>>&& promise) const
+{
+    m_private->getPhotoSettings()->whenSettled(RunLoop::main(), [protectedThis = Ref { *this }, promise = WTFMove(promise)] (RealtimeMediaSource::PhotoSettingsPromise::Result&& result) mutable {
```

Comment:
Reusing PhotoSettingsPromise names for different types in in RealtimeMediaSource and ImageCapture makes it slightly confusing.
I would suggest using auto&& here if/when we can.

RealtimeMediaSource::PhotoSettingsPromise::Result would ideally be an ExceptionOr<PhotoSettings>.
It would have been nice to use ExceptionOr<PhotoSettings> to be able to simply write: `promise.settle(WTFMove(result));`
But ExceptionOr is not in platform.

Reply:
I agree. I renamed the later `PhotoSettingsNativePromise` and switched to `auto` where possible.